DNN Platform/DotNetNuke.Web/InternalServices/ControlBarController.cs

```diff
@@ -2,914 +2,919 @@
+                                if (!string.IsNullOrEmpty(content))
+                                {
+                                    ((IPortable)objObject).ImportModule(newModule.ModuleID, content, newModule.DesktopModule.Version, userID);
+                                }
+                            }
+                            finally
+                            {
+                                SetCloneModuleContext(false);
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    // copy tab module settings
+                    newModule.TabModuleSettings.Clear();
+                    foreach (var key in moduleInfo.TabModuleSettings.Keys)
+                    {
+                        newModule.TabModuleSettings.Add(key, moduleInfo.TabModuleSettings[key]);
+                    }
+
+                    ModuleController.Instance.AddModule(newModule);
+                }
+
+                // if the tab of original module has custom stylesheet defined, then also copy the stylesheet
+                // to the destination tab if its custom stylesheet is empty.
+                var originalTab = TabController.Instance.GetTab(moduleInfo.TabID, moduleInfo.PortalID);
+                var targetTab = PortalSettings.Current.ActiveTab;
+                if (originalTab != null
+                    && originalTab.TabSettings.ContainsKey("CustomStylesheet")
+                    && !string.IsNullOrEmpty(originalTab.TabSettings["CustomStylesheet"].ToString())
+                    && (!targetTab.TabSettings.ContainsKey("CustomStylesheet") ||
+                        string.IsNullOrEmpty(targetTab.TabSettings["CustomStylesheet"].ToString())))
+                {
+                    TabController.Instance.UpdateTabSetting(targetTab.TabID, "CustomStylesheet", originalTab.TabSettings["CustomStylesheet"].ToString());
+                }
+
+                if (remote)
+                {
+                    // Ensure the Portal Admin has View rights
+                    var permissionController = new PermissionController();
+                    ArrayList arrSystemModuleViewPermissions = permissionController.GetPermissionByCodeAndKey("SYSTEM_MODULE_DEFINITION", "VIEW");
+                    this.AddModulePermission(
+                        newModule,
+                        (PermissionInfo)arrSystemModuleViewPermissions[0],
+                        PortalSettings.Current.AdministratorRoleId,
+                        Null.NullInteger,
+                        true);
+
+                    // Set PortalID correctly
+                    newModule.OwnerPortalID = newModule.PortalID;
+                    newModule.PortalID = PortalSettings.Current.PortalId;
+                    ModulePermissionController.SaveModulePermissions(newModule);
+                }
+                else

```

Comment:
The `if` of this `else` has logic to set view permissions if `remote` is true, just thinking here if maybe we should just remove that condition would be the bug fix. Because now if the copy is on the same site it will copy all permissions (which users might not be expecting to happen). I think if at least the admin sees the module it would be good enough to fix the reported issue IMO.